### PR TITLE
working with erobhal

### PR DESCRIPTION
**Pull request by Phil-Friderici**
Moving the logic into the manifest.

Not finished, needs more spec test, docs and style fixes.

### CODE REVIEW

**Comment by Phil-Friderici** [comment]
@ghoneycutt This patch brings in support for periodic tasks on Suse and RHEL5 that was missing (removed actually) from the vendor versions of crontab.

My question here is if I should activate that functionality by default or not ?
pro: it is a fix for a problem in my eyes
con: it will change the default behaviour of the module

What do you think ?

File: spec/fixtures/custpath_crontab-RedHat-5 [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# .---------------- minute (0 - 59)
+# |  .------------- hour (0 - 23)
+# |  |  .---------- day of month (1 - 31)
+# |  |  |  .------- month (1 - 12) OR jan,feb,mar,apr ...
+# |  |  |  |  .---- day of week (0 - 6) (Sunday=0 or 7) OR sun,mon,tue,wed,thu,fri,sat
+# |  |  |  |  |
+# *  *  *  *  * user-name command to be executed
+
+# run-parts
+01 * * * * root run-parts /path/to/hourly
```
Comment (ghoneycutt) [inline spec/fixtures/custpath_crontab-RedHat-5]: should `/path/to/hourly` be `/etc/cron.hourly` or something similar?

Comment (Phil-Friderici) [inline spec/fixtures/custpath_crontab-RedHat-5]: This is testing the possibility to change the path to the periodic directory

**Comment by Phil-Friderici** [comment]
updated spec & docs

**Comment by Phil-Friderici** [comment]
@erobhal From my perspective these commits are ready to get merged. It should fully feature your functional change with some additions from my side. Please give it a test that I haven't missed something you need.

**Comment by ghoneycutt** [comment]
I don't have strong feelings about it. Personally I don't use the periodic entries and I think it was right for the distro's to remove them in favor of using cron.d. Adding them to systems where they don't already exist could cause confusion.

Perhaps if certain platforms use it, then it could be a boolean to be enabled for those platforms.

File: manifests/init.pp [hunk 2]
```diff
@@ -58,8 +60,15 @@
     'Debian': {
       $package_name_default = 'cron'
       $service_name_default = 'cron'
+      $periodic_jobs_content_default = []
     }
     'Suse': {
+      $periodic_jobs_content_default = [
```
Comment (ghoneycutt) [inline manifests/init.pp]: This might be simpler if it was just a string with `\n` in it. Not sure that using an array buys you anything.

**Comment by ghoneycutt** [comment]
Which after refreshing the code, seems to be what you did:)

**Comment by erobhal** [comment]
Problem is that several packages depend on them. Like logrotate for example. We had a bunch of SuSE-systems where the logs didn't get rotated.

File: manifests/init.pp (hunk 2 above)
Comment (Phil-Friderici) [inline manifests/init.pp]: I prefer arrays for better readability of the content. I respect people who prefer using strings in line 132 and it's tested in the specs (line 406). Maybe I should add a hint to the docs too.

**Comment by Phil-Friderici** [comment]
> I don't have strong feelings about it. Personally I don't use the periodic entries and I think it was right for the distro's to remove them in favor of using cron.d. Adding them to systems where they don't already exist could cause confusion.

> Perhaps if certain platforms use it, then it could be a boolean to be enabled for those platforms.

I read the OS vendor docs to see what their defaults looks like. Only RHEL5 and Suse are using them and so does the module now by default.

> Which after refreshing the code, seems to be what you did:)

And backward compatibility can be reached by setting `$periodic_jobs_manage ` to false.

**Gentlemen, thanks for your help and time !!** I will add the missing string support for the content, merge, and release it.

**Comment by Phil-Friderici** [comment]
Released in 2.1.0

**Pull request merged**